casa.wlk

```diff
@@ -1,4 +1,66 @@
-import cosas.* 
+import cosas.*
+import cuentas.*
 
 object casaDePepeYJulian {
-}
+  const cosasCompradas = []
+  var dineroGastado = 0
+  var property cuentaBancaria = cuentaCorriente
+  
+  method validarListaVacia(unaLista) {
+    if (unaLista.isEmpty()) self.error("la lista no puede estar vacia")
+  }
+  
+  method cosasCompradas() = cosasCompradas
+  
+  method dineroGastado() = dineroGastado
+  
+  method saldo() = self.cuentaBancaria().saldo()
+  
+  method comprar(cosa) {
+    cuentaBancaria.extraer(cosa.precio())
+    dineroGastado += cosa.precio()
+    cosasCompradas.add(cosa)
+  }
+  
+  method cantidadDeCosasCompradas() = cosasCompradas.size()
+  
+  method tieneAlgun(categoria) = cosasCompradas.any(
+    { cosa => cosa.categoria() == categoria }
+  )
+  
+  method ultimaCompra() {
+    self.validarListaVacia(cosasCompradas)
+    return cosasCompradas.last()
```

Comment:
No es necesario hacer la validacion de lista vacía, ya que si está vacía el método last se va a encargar de romper por vos. Sólo tendría sentido hacer esta validación si vos querés enviar un mensaje de error más detallado del problema.  algo como "No existen compras" en lugar de "index of error".